src/watchmaker/status/__init__.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+"""Status Module."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals, with_statement)
+
+import logging
+
+import watchmaker.config.status as status_config
+from watchmaker.status.providers.abstract import AbstractStatusProvider
+from watchmaker.status.providers.aws import AWSStatusProvider
+from watchmaker.status.providers.azure import AzureStatusProvider
+from watchmaker.utils.imds.detect import provider
+
+
+class Status():
+    """Status factory for providers."""
+
+    _PROVIDERS = {
+        AWSStatusProvider.identifier: AWSStatusProvider,
+        AzureStatusProvider.identifier: AzureStatusProvider
+    }
+
+    def __init__(self, config=None, logger=None):
+        self.logger = logger or logging.getLogger(__name__)
+        self.status_providers = {}
+        self.targets = {}
+        self.initialize(config)
+
+    def initialize(self, config=None):
+        """Initialize status providers."""
+        status_provider_ids = []
+        if status_config.get_cloud_identifiers(config):
+            cloud_identifier = provider()
+            if cloud_identifier != AbstractStatusProvider.identifier:
+                status_provider_ids.append(cloud_identifier)
+
+        status_provider_ids += \
+            status_config.get_non_cloud_identifiers(config)
+
+        self.status_providers = \
+            self.__get_status_providers(status_provider_ids)
+
+        for k, _v in self.status_providers:  # pylint: disable=invalid-name
```

Comment:
Since you aren't using the value...

```suggestion
        for provider in self.status_providers:
```